### ISSUE TITLE
ElastAlert Kibana Plugin - Undefined [object Object]

### ISSUE THREAD

**Issue by janjaom** [bitsensor/elastalert-kibana-plugin#162]
Anyone issue this?
![ElastAlert](https://user-images.githubusercontent.com/65362239/101392116-1bf89500-38a4-11eb-8189-658655587ec9.JPG)

**Comment by nsano-rururu** [comment]
Have you performed the following operations before starting?

```
mkdir -p rules rule_templates
chmod -R 777 rules rule_templates
```

**Comment by nsano-rururu** [comment]
- What is the version of kibana?
- Direct installation? Or docker?

**Comment by nsano-rururu** [comment]
docker compose sample
https://github.com/nsano-rururu/elastalert-kibana-plugin/wiki/docker-compose-sample

**Comment by janjaom** [comment]
> Have you performed the following operations before starting?
> 
> ```
> mkdir -p rules rule_templates
> chmod -R 777 rules rule_templates
> ```

Not yet...

**Comment by janjaom** [comment]
> * What is the version of kibana?
> * Direct installation? Or docker?

Kibana version: 7.9.3
Direct Installation

**Comment by nsano-rururu** [comment]
Have you added settings to kibana.yml?
https://github.com/bitsensor/elastalert-kibana-plugin#configuration

**Comment by janjaom** [comment]
> Have you added settings to kibana.yml?
> https://github.com/bitsensor/elastalert-kibana-plugin#configuration

Yep! Actually my ElastAlert Server running in same local Elastic Stack! I have added ip local and port default elasticsearch (9200)

**Comment by janjaom** [comment]
elastalert-kibana-plugin.serverHost: (ip local elastic stack)
elastalert-kibana-plugin.serverPort: (9200 -  Port ElasticSearch Local)

**Comment by nsano-rururu** [comment]
Please specify 3030 because it is the port of elastalert-server.

**Comment by nsano-rururu** [comment]
You can see it by looking at the following settings.
https://github.com/nsano-rururu/elastalert-kibana-plugin/wiki/docker-compose-sample

**Comment by janjaom** [comment]
> Please specify 3030 because it is the port of elastalert-server.

I don't use docker for this! My ElastAlert Server is running perfect on background... I Just would like see API Plugin on Kibana and don't have port like 3030 for Elastalert-server

**Comment by nsano-rururu** [comment]
Read the document. Do not be silly!

**Comment by nsano-rururu** [comment]
Do you still use this? You are!

Line 3,4
https://github.com/bitsensor/elastalert/blob/master/config/config.json
```
  "port": 3030,
  "wsport": 3333,
```

**Comment by nsano-rururu** [comment]
I have provided the information I know. I'll do the rest myself. goodbye

**Comment by nsano-rururu** [comment]
Be aware of the contents of the configuration file and which port to use when it actually works.

**Comment by nsano-rururu** [comment]
It's bad not to look at the contents of the configuration file.

**Comment by nsano-rururu** [comment]
Deadly

**Comment by nsano-rururu** [comment]
3333 of wsport should be the Port number of WebSocket.

**Comment by nsano-rururu** [comment]
> I don't use docker for this! My ElastAlert Server is running perfect on background... I Just would like see API Plugin on Kibana and don't have port like 3030 for Elastalert-server

How to write the configuration file is helpful. ElastAlertServer works on ports 3030 and 3333. I don't know why.

**Comment by janjaom** [comment]
@nsano-rururu I have been searching and has been differents path installation elastalert, my installation is from Yelp/ElastAlert and not bitsensor/ElastAlert,

**Comment by nsano-rururu** [comment]
elastalert-kibana-plugin uses ElastAlertServer (node.js) to operate ElastAlert(Python).

**Comment by nsano-rururu** [comment]
Please read the Praeco part of the architecture diagram below as ElastAlert-Kibana-Plugin.
https://user-images.githubusercontent.com/611996/52892144-90a19300-3155-11e9-8050-cb4a440411a4.png

**Comment by nsano-rururu** [comment]
ElastAlert-Kibana-Plugin cannot be used when only Yelp / ElastAlert is installed.

**Comment by nsano-rururu** [comment]
> @nsano-rururu I have been searching and has been differents path installation elastalert, my installation is from Yelp/ElastAlert and not bitsensor/ElastAlert,

I found that I couldn't trust what you were saying. You said ElastAlertServer is working. This is terrible.

**Comment by nsano-rururu** [comment]
The question should have been resolved so please close it

**Comment by nsano-rururu** [comment]
Ask a question in the repository you are maintaining
https://github.com/Karql/elastalert-kibana-plugin
